.github/workflows/codeql.yml

```diff
@@ -16,28 +16,25 @@ jobs:
     name: Analyze ${{ matrix.language }}
     runs-on: ubuntu-latest
     permissions:
-      actions: read
-      contents: read
       security-events: write
 
     strategy:
       fail-fast: false
       matrix:
-        language: [python, cpp]
+        language: ["cpp", "python"]
 
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
         with:
           submodules: recursive
           fetch-depth: 0
 
+      - name: Set up mold as linker
+        uses: rui314/setup-mold@v1
+
       - name: Install Dependencies (python)
         if: ${{ matrix.language == 'python' }}
-        run: |
-          sudo apt-get update
-          sudo apt-get install --yes python3-pip
-          pip3 install qiskit-terra
+        run: pip3 install qiskit-terra
 
       - name: Configure CMake
```

Comment:
I am fairly sure that this step should happen after the CodeQL `init` action since that action sets quite some environment variables and other things.

Furthermore, why does this try to also build the QFR and DDPackage tests? Seems quite wasteful to me, since all of that is covered in the respective repositories already.